src/test/java/com/systelab/seed/patient/PatientResourceTest.java

```diff
@@ -179,61 +172,68 @@ public void testGetPatientList() {
         long initialSize = patientsBefore.getTotalElements();
         savePatientsDatabase(patientsBefore.getContent());
 
         createSomePatients(numberOfPatients);
 
-        PatientsPage patientsAfter = given()
-                .when().get("/patients")
-                .then().assertThat().statusCode(200)
-                .extract().as(PatientsPage.class);
+        Response responseAfter = doGetPatientList();
+        PatientsPage patientsAfter = responseAfter.then().assertThat().statusCode(200).extract().as(PatientsPage.class);
         long finalSize = patientsAfter.getTotalElements();
         savePatientsDatabase(patientsAfter.getContent());
 
         TestUtil.checkANumber("List size", initialSize + numberOfPatients, finalSize);
     }
 
+
     @Description("Get a Patient by id")
     @Test
     public void testGetPatient() {
 
-        Patient patient = getPatientData("John", "Burrows", "jburrows@werfen.com");
-        Patient patientCreated = given().body(patient)
-                .when().post("/patients/patient")
-                .then().assertThat().statusCode(200)
-                .extract().as(Patient.class);
+        Patient patient = getPatientData("John", "Burrows", "jburrows@werfen.com","123456", null,null,null, null);
+        Response responseCre = doCreatePatient(patient);
+        Patient patientCreated = responseCre.then().assertThat().statusCode(200).extract().as(Patient.class);
 
+        UUID patientCreatedId = patientCreated.getId();
         TestUtil.checkObjectIsNotNull("Patient ID " + patientCreated.getId(), patientCreated.getId());
-        Patient patientRetrieved = given()
-                .when().get("/patients/" + patientCreated.getId())
-                .then().assertThat().statusCode(200)
-                .extract().as(Patient.class);
+
+        Response responseGet = doGetPatient(patientCreatedId);
+        Patient patientRetrieved = responseGet.then().assertThat().statusCode(200).extract().as(Patient.class);
         Assertions.assertNotNull(patientRetrieved, "Patient not retrieved");
-        if (patientRetrieved != null) {
-            TestUtil.checkField("Name", "John", patientRetrieved.getName());
-            TestUtil.checkField("Surname", "Burrows", patientRetrieved.getSurname());
-            TestUtil.checkField("Email", "jburrows@werfen.com", patientRetrieved.getEmail());
 
-            TestUtil.checkField("Medical Number", "123456", patientRetrieved.getMedicalNumber());
-            TestUtil.checkField("DOB", LocalDate.of(1948, 8, 11), patientRetrieved.getDob());
-        }
+        if (patientRetrieved != null)
+            checkPatientData(patientCreated, patientRetrieved);
+    }
+
+     @Description("Get an invalid Excel file with patients")
+    @Test
+    public void testGetPatientsInvalidExcel() {
+
+    int statusCode = given()
+        .when().post("/patients/report")
+        .then()
+        .extract().statusCode();
+        TestUtil.checkField("Status Code", 400, statusCode);
```

Comment:
This test is failing: you are expecting status 400, but in the last runs the code received is 405.